class.jetpack.php

```diff
@@ -6160,7 +6160,18 @@ public static function validate_sync_error_idc_option() {
 			$local_options = self::get_sync_error_idc_option();
 			// Ensure all values are set.
 			if ( isset( $sync_error['home'] ) && isset( $local_options['home'] ) && isset( $sync_error['siteurl'] ) && isset( $local_options['siteurl'] ) ) {
-				if ( $sync_error['home'] === $local_options['home'] && $sync_error['siteurl'] === $local_options['siteurl'] ) {
+
+				// If the WP.com expected home and siteurl match local home and siteurl it is not valid IDC.
+				if (
+						isset( $sync_error['wpcom_home'] ) &&
+						isset( $sync_error['wpcom_siteurl'] ) &&
+						$sync_error['wpcom_home'] === $local_options['home'] &&
+						$sync_error['wpcom_siteurl'] === $local_options['siteurl']
+				) {
+					$is_valid = false;
+					// Enable migrate_for_idc so that sync actions are accepted.
+					Jetpack_Options::update_option( 'migrate_for_idc', true );
```

Comment:
Do you mind providing a TL;DR on why the `migrate_for_idc` is needed to be set to true here? Is it because the remote counterpart still thinks that the site is in IDC mode?

Reply:
The production site is sending traffic to WP.com using a different domain. Without migrate_for_idc being set the site will continue to trigger idc issues. The new routine would clear when a request from the expected domain but WP.com would deny the sync request from the other domain and could lead to lag and other issues. 

This allows WP.com to process sync data from the alternate domain which would be needed if cron was sending as a different domain and the site was configured to only send sync on cron.